### ISSUE TITLE
Replace hint with `print_with_newline` deletes semicolon

### ISSUE THREAD

**Issue by jonasbb** [rust-lang-nursery/rust-clippy#2648]
Given the simple source code

```rust
fn main() {
    println!("");
}
```

clippy (Version 0.0.193) creates the following error:

```
warning: using `println!("")`
 --> src/main.rs:2:5
  |
2 |     println!("");
  |     ^^^^^^^^^^^^^ help: replace it with: `println!()`
  |
  = note: #[warn(print_with_newline)] on by default
```

The "replace it with" hint falsely removes the semicolon. Note how the `^` include the semicolon but not the replacement text. This is especially bad if you use automatic fixes as provided with RLS.

Side-note: I find if pretty confusing that the newline-lint is responsible for linting this, because no newline is involved.
